src/dom.js

```diff
@@ -1,37 +1,40 @@
 import { SVG } from './reconciler'
 
-function updateProperty (dom, name, value, newValue) {
-  if (name === 'style') {
-    for (let k in value) if (!newValue[k]) dom[name][k] = ''
-    for (let k in newValue) dom[name][k] = newValue[k]
-  } else if (name[0] === 'o' && name[1] === 'n') {
-    name = name.slice(2).toLowerCase()
-    if (value) dom.removeEventListener(name, value)
-    dom.addEventListener(name, newValue)
-  } else if (name in dom && !(dom instanceof SVGElement)) {
-    dom[name] = newValue == null ? '' : newValue
-  } else if (newValue == null || newValue === false) {
-    dom.removeAttribute(name)
-  } else {
-    dom.setAttribute(name, newValue)
-  }
-}
+export function updateElement (dom, oldProps, newProps) {
+  const names = Object.keys({...oldProps, ...newProps})
+
+  for (let name of names) {
+    if (name === "children") {
+      continue
+    }
 
-export function updateElement (dom, props, newProps) {
-  Object.keys(newProps)
-    .filter(isNew(props, newProps))
-    .forEach(key => updateProperty(dom, key, props[key], newProps[key]))
+    const oldValue = oldProps[name]
+    const newValue = newProps[name]
+
+    if (name === 'style') {
+      for (let k in oldValue) if (!newValue[k]) dom[name][k] = ''
```

Comment:
should be `oldValue || {}` maybe? to make sure old style-properties get removed?

Reply:
emmm
The `in` operator of `for` loop would judge first in fact, there is no errors which will be throwed even if `oldValue` is `undefind` here. But It was not rigorous or safe enough.

maybe should be 
```js
    const oldValue = oldProps[name] || {}
    const newValue = newProps[name] || {}
``` 
here ?  
@mindplay-dk

Reply:
That would break all the checks below it - `if (newValue == null)` and so on.